js/stdlib/ts/ETH_like.ts

```diff
@@ -319,19 +319,40 @@ const makeHasLogFor = ( getCtcAddress: (() => Address), iface:Interface, i:numbe
 
 const { randomUInt, hasRandom } = makeRandom(32);
 
-const balanceOf = async (acc: Account | Address, token: Token|false = false): Promise<BigNumber> => {
-  let addressable = (typeof acc == 'string') ? acc : acc.networkAccount;
-  if (!addressable) {
-    throw Error(`Cannot get the address of: ${acc}`);
-  }
-  return balanceOfNetworkAccount(addressable, token);
-};
-
 const minimumBalanceOf = async (acc: Account | Address): Promise<BigNumber> => {
   void acc;
   return zeroBn;
 };
 
+const balanceOf = async (acc: Account | Address, token?: Token): Promise<BigNumber> => {
+  return (await balancesOf(acc, [token || null]))[0];
+};
+
+const balancesOf = async (acc: Account | Address, tokens: Array<Token|null>): Promise<Array<BigNumber>> => {
+  // Implementation stolen from old balanceOf and balanceOfNetworkAccount
+
+  // Address = string. If given an Address, use it as the address and networkAccount.
+  // If given an Account, extract the address and networkAccount from it.
+  const [address, networkAccount] = (typeof acc == 'string')
+                                  ? [acc, { address: acc }]
+                                  : [await getAddr(acc), acc.networkAccount];
+
+  const balanceOfSingleToken = async (token: Token | null) => {
```

Comment:
This implementation is bad. You should leave the old `balanceOf` function exactly as it is and make it so that this function `balancesOf` just calls it with mapping and `Promise.all`